core/src/Kokkos_SYCL_Space.hpp

```diff
@@ -187,6 +191,45 @@ struct MemorySpaceAccess<Kokkos::Experimental::SYCLSharedUSMSpace,
   enum : bool { deepcopy = true };
 };
 
+template <>
+struct MemorySpaceAccess<
+    Kokkos::Experimental::SYCLDeviceUSMSpace,
+    Kokkos::ScratchMemorySpace<Kokkos::Experimental::SYCL>> {
```

Comment:
I have chosen this form since I would run into circular dependencies otherwise (`Kokkos::Experimental::SYCL_*_Space` is used in `Kokkos::Experimental::SYCL`).